src/acceleration/AitkenAcceleration.cpp

```diff
@@ -117,13 +118,16 @@ void AitkenAcceleration::iterationsConverged(
 }
 
 void AitkenAcceleration::concatenateCouplingData(
-    const DataMap &cplData, const std::vector<DataID> &dataIDs, Eigen::VectorXd &targetValues, Eigen::VectorXd &targetOldValues, double windowStart) const
+    const DataMap &cplData, const std::vector<DataID> &dataIDs, Eigen::VectorXd &targetValues, Eigen::VectorXd &targetOldValues, double windowStart, double windowEnd) const
 {
   Eigen::Index offset = 0;
+
   for (auto id : dataIDs) {
-    Eigen::Index size      = cplData.at(id)->values().size();
-    auto &       values    = cplData.at(id)->values();
-    const auto & oldValues = cplData.at(id)->previousIteration();
+    Eigen::Index size = cplData.at(id)->values().size();
+
+    Eigen::VectorXd values    = cplData.at(id)->timeStepsStorage().sample(windowEnd);
```

Comment:
```suggestion
    Eigen::VectorXd values    = cplData.at(id)->timeStepsStorage().sample(windowEnd).values();
```
This was changed very recently via #2190

Reply:
Actually, prefer to save the result of sample and then use `result.values()` in the computation. This can avoid copies.